app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -310,11 +317,13 @@ class SnapKeyringImpl implements SnapKeyringCallbacks {
         // state, so we can safely uses this state to run post-processing.
         // (e.g. renaming the account, select the account, etc...)
 
-        // Set the selected account to the new account
-        this.#messenger.call(
-          'AccountsController:setSelectedAccount',
-          accountId,
-        );
+        if (!skipSetSelectedAccountStep) {
+          // Set the selected account to the new account
+          this.#messenger.call(
+            'AccountsController:setSelectedAccount',
+            accountId,
+          );
+        }
```

Comment:
This is a new option which will be required for some custom flow like the bridge one (e.g. creating a Solana account on the fly and bridge from an EVM account)